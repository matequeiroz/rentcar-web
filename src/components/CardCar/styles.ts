import styled from "@emotion/styled";

type InfoContentProps = {
  price?: boolean;
};

export const Container = styled.section`
  background-color: var(--white-100);
`;

export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 2.4rem;
`;

export const InfoTitle = styled.h6`
  font-size: 1rem;
  line-height: 1rem;
  color: var(--gray-300);
  text-transform: uppercase;
`;
export const InfoContent = styled.h5<InfoContentProps>`
  font-size: 1.5rem;
  line-height: 1.6rem;
  color: ${(props) => (props.price ? `var(--red-900)` : `var(--gray-600)`)};
  padding-top: 0.4rem;
`;

export const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
`;
