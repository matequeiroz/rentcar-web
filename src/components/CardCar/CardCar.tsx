import Image from "next/image";

import AudiSvg from "../../../public/images/cars/Audi.svg";
import * as S from "./styles";

export default function CardCar() {
  return (
    <S.Container>
      <S.Content>
        <S.InfoContainer>
            <div>
                <S.InfoTitle>Audi</S.InfoTitle>
                <S.InfoContent>RS 5 Coup√©</S.InfoContent>
            </div>
            <div>
                <S.InfoTitle>Ao dia</S.InfoTitle>
                <S.InfoContent price>R$ 120</S.InfoContent>
            </div>
        </S.InfoContainer>
        <div>
          <Image src={AudiSvg} alt="Car photo" />
        </div>
      </S.Content>
    </S.Container>
  );
}
