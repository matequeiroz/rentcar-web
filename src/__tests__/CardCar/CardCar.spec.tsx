import { render } from "@testing-library/react";

import CardCar from "@/components/CardCar/CardCar";
import Audi from "../../../public/images/cars/Audi.svg";

describe("CardCar.tsx", () => {
  const mock = {
    brand: "Audi",
    carName: "RS 5 CoupÃ©",
    priceToDay: 190,
    photoUrl: Audi,
  };

  it("should render the component", () => {
    const { container } =render(
      <CardCar
        brand={mock.brand}
        carName={mock.carName}
        photoUrl={mock.photoUrl}
        priceToDay={mock.priceToDay}
      />
    );

    expect(container).toMatchSnapshot()
  });
});
