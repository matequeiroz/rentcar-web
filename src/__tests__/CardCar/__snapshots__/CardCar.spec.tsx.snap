// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCar.tsx should render the component 1`] = `
<div>
  <section
    class="css-nium2g"
    data-testid="CardCar-component"
  >
    <div
      class="css-1lse5tz"
    >
      <div
        class="css-z6kccy"
      >
        <div>
          <h6
            class="css-9qeut1"
          >
            Audi
          </h6>
          <h5
            class="css-k3gmv8"
          >
            RS 5 Coup√©
          </h5>
        </div>
        <div>
          <h6
            class="css-9qeut1"
          >
            Ao dia
          </h6>
          <h5
            class="css-2t47hf"
          >
            R$ 
            190
          </h5>
        </div>
      </div>
      <div>
        <img
          alt="Car photo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="100"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="180"
        />
      </div>
    </div>
  </section>
</div>
`;
